lib/common/ipc_client.c

```diff
@@ -1200,16 +1200,21 @@ crm_ipc_name(crm_ipc_t * client)
 // \return Standard Pacemaker return code
 static int
 internal_ipc_get_reply(crm_ipc_t *client, int request_id, int ms_timeout,
-                       ssize_t *bytes)
+                       ssize_t *bytes, xmlNode **reply)
 {
     pcmk__ipc_header_t *hdr = NULL;
-    xmlNode *reply = NULL;
-    time_t timeout = time(NULL) + 1 + pcmk__timeout_ms2s(ms_timeout);
-    int32_t qb_timeout = 1000;
+    xmlNode *xml = NULL;
+    time_t timeout = 0;
+    int32_t qb_timeout = -1;
     int rc = pcmk_rc_ok;
 
+    if (ms_timeout != 0) {
```

Comment:
This isn't right. Check whether it's greater than 0. Any negative value will get cast to a `guint`. If it's `-1`, then it probably won't make much difference, because it will get cast to `UINT_MAX`.

The 0 case itself is weird and sort of undefined. Any negative value is supposed to mean no timeout. 0 is documented to mean a 5-second timeout, but I don't think that's actually the case.